### PR TITLE
Error catching

### PR DESCRIPTION
**Pull request by perryr16**
* catch if arrays are empty
* catch if no data is returned

### CODE REVIEW

File: poros/game_results.rb [hunk 1]
```diff
@@ -22,11 +23,13 @@ def game_params_by_name(game_name)
...
       screenshots:  format_screenshots(params), # array of ids
       themes:       run_array(:get_game_themes, :name, params[:themes]), # array of ids
-      videos:       "https://www.youtube.com/watch?v=#{youtube_id(params[:videos][0])}" # good -> array of ids
+      # video:       "https://www.youtube.com/watch?v=#{youtube_id(params[:videos][0])}" # good -> array of ids
+      video:        "#{youtube_id(params[:videos][0])}" # good -> array of ids
     }}
   end
 
   def run_array(method, key, array)
+    return nil if !array.is_a?(Array)
```
Comment (whitneykidd) [inline poros/game_results.rb:32]: I like this work around

File: poros/game_results.rb [hunk 2]
```diff
@@ -6,6 +6,7 @@ def service
   end
 
   def game_params_by_name(game_name)
+    return "Invalid Game Name" if service.get_games_by_name(game_name).empty?
```
Comment (perryr16) [inline poros/game_results.rb:9]: if game name is not a perfect match, we can now `fail gracefully`.  The goal all along

File: poros/game_results.rb [hunk 3]
```diff
@@ -22,11 +23,13 @@ def game_params_by_name(game_name)
       similars:     run_array(:get_games_by_id, :name, params[:similar_games]), # array of ids
       screenshots:  format_screenshots(params), # array of ids
       themes:       run_array(:get_game_themes, :name, params[:themes]), # array of ids
-      videos:       "https://www.youtube.com/watch?v=#{youtube_id(params[:videos][0])}" # good -> array of ids
+      # video:       "https://www.youtube.com/watch?v=#{youtube_id(params[:videos][0])}" # good -> array of ids
+      video:        "#{youtube_id(params[:videos][0])}" # good -> array of ids
```
Comment (perryr16) [inline poros/game_results.rb:27]: pass video id, not whole video

File: poros/game_results.rb (hunk 1 above)
Comment (perryr16) [inline poros/game_results.rb:32]: Dear Microservice,
If its not an array, dont panic.

**Review by perryr16 — COMMENTED** [review]
microservice error catching

**Pull request merged**